README.md

```diff
@@ -1,45 +1,272 @@
-1. It will allow you to set branch protections on the `default` branch no matter what it is called by calling it `default` in `.github/settings.yml`.
+It is loosely based on [Settings Probot](https://github.com/probot/settings) GitHub App. However, it differs from the original probot settings app in several important ways:
 
-## Usage
+1. In `safe-settings` all the settings are stored centrally in an `admin` repo within the organization. 
+1. There are 3 levels at which the settings could be managed:
+   1. Org-level settings are defined in `.github/settings.yml` 
+   1. The org-level settings could be overridden at a  `suborg` level. These would be a collection of repos belonging to a project or a business unit, or a team. The `suborg`settings reside in the `.github/suborgs`folder.
+   1. Lastly, we could override settings for individual repos by creating a repo specific yaml in `.github/repos`folder
+1. `Safe-settings` explicitly looks in the `admin` repo in the organization for the settings files. The `admin` repo should be a restricted repository with `branch protections` and `codeowners` . 
+1. To address the scalability concerns for large organizations having thousands of repos, it is recommended to break the settings into org-level, suborg-level, and repo-level units. This will allow different teams to be define and manage policies for their specific projects or business units.
+
+
+
+## Table of Contents
+[TOC]
+
+
+## How it works
+
+The App listens to the following webhook events and does the following:
+
+- **push**: If settings are created or settings are modified, that is, if  push happens in the `default` branch of the `admin` repo and the file added or changed is `.github/settings.yml` or `.github/repos/*.yml`or `.github/suborgs/*.yml`, then the settings would be applied either globally to all the repos, or specific repos. For each repo the settings that would be applied depend on the default settings for the org, overlayed with settings for the suborg that the repo belongs to, overlayed with the settings for that specific repo.
+  
+- **repository.created**: If a repository is created in the org, the settings for the repo - the default settings for the org, overlayed with settings for the suborg that the repo belongs to, overlayed with the settings for that specific repo - is applied. To prevent `infinite loop` if the repo created event is trigged by itself(bot), it would be ignored. This ensures that the policy is enforced even to nee repositories.
+
+- **repository.edited**: If the `default` branch is changed if the the settings that would be applied for the repo
+
+**Exceptions:** Certain repos may be exempted by specifying them in a runtime config file called `depolyment-settings.yml`. If no file is specified, then the following repositories -  `'admin', '.github', 'safe-settings'` are exempted by default.
```

Comment:
Would it be correct to state the location as the following:
```suggestion
**Exceptions:** Certain repos may be exempted by specifying them in a runtime config file called `/.github/depolyment-settings.yml`. If no file is specified, then the following repositories -  `'admin', '.github', 'safe-settings'` are exempted by default.
```

Reply:
Given the generic name `admin` for the repo I would encourage any/all configurations be more clearly related to safe-settings through subdirectory usage.